iwyu_output.cc

```diff
@@ -1426,6 +1426,70 @@ void CalculateIwyuForFullUse(OneUse* use,
   }
 }
 
+// This function removes all OneIncludeOrForwardDeclareLines a that are desired
+// but not present but transitively included by some
+// OneIncludedOrForwardDeclareLine b that is desired and present and adds a's
+// symbol counts to b.
+vector<OneIncludeOrForwardDeclareLine> PruneTransitivelyPresent(
+    const vector<OneIncludeOrForwardDeclareLine> & lines,
+    const IwyuPreprocessorInfo* preprocessor_info) {
+  // In the first step, we copy all lines that are an include line, not present,
+  // desired, and whose include_file is transitively included by another line
+  // which is a desired include line into a range second and the remaining ones
+  // into another range first.
+  // This is because elements of range second are then precisely those whose
+  // symbols are already exposed through a different header that will is to be
+  // included anyway, so they can be omitted.
+  const auto is_transitively_present = [&lines, preprocessor_info](
```

Comment:
I wonder if this name is right. I realize I suggested it, but I meant `is_transitively_present` as an example to make you describe the purpose. It looks to me like it now returns `false` for lines that have transitive representation but `true` for everything else, so is it backwards?

I'm also having trouble with the `source` and `target` terminology. I think `target` could be renamed `line` and `source` could be renamed `other`. That way it's clearer that you cross-reference every include-line against every other.

Reply:
You're right, I flipped its logic, so that it does match the name `is_trasitively_present` now and accordingly switched `first` and `second`.  (And then renamed `first` and `second`.)